in-progress/8757-fees/design.md

```diff
@@ -0,0 +1,192 @@
+| Owners               | @just-mitch @LHerskind                                             |
+| Approvers            | @Maddiaa0 @PhilWindle @dbanks12 @nventuro @joeandrews @aminsammara |
+| Target Approval Date | 2024-10-24                                                         |
+
+## Mana and Costs
+
+Refer to the unit of work on L2 as "mana", analogous to gas on L1.
+
+A transaction incurs the following costs:
+1. Proposers/validators simulate the transaction
+2. Proposers publish pending blocks to L1
+3. Provers generate the avm proofs for the transaction
+4. Provers generate the rollup proofs for the blocks/epoch
+5. Provers verify the epoch proof on L1
+
+Some costs are independent of the computation or data of the transaction itself, specifically numbers 2, 4, and 5; regardless, they must be covered by the transaction.
+
+## Protocol Defined Constants
+
+- `overhead_mana_per_tx` - the overhead cost in mana for a transaction
+- `target_mana_per_block` - the amount of mana that an "average" block is expected to consume
+- `blobs_per_block` - the number of blobs per block that are compensated for in the base fee
+- `l2_slots_per_l2_epoch` - the number of L2 slots in each L2 epoch
+- `l1_gas_per_block_proposed` - the amount of L1 gas required to propose an L2 block on L1
+- `l1_gas_per_epoch_verified` - the amount of L1 gas required to verify an L2 epoch on L1
+- `minimum_proving_cost_per_mana` - the minimum cost in wei for proving a unit of mana
+- `maximum_proving_cost_multiplier_per_block` - the maximum percentage increase in the cost of proving a unit of mana per block
+- `minimum_fee_asset_price` - the minimum price in wei of the fee asset (e.g. TST)
+- `maximum_fee_asset_price_multiplier_per_block` - the maximum percentage increase in the price of the fee asset per block
+- `underlying_base_fee_oracle_update_interval` - the minimum number of slots between updates to the underlying base fee oracle
+- `congestion_factor_multiplier` - the constant factor to multiply the congestion factor by
+- `maximum_epoch_proof_quote_fee` - the maximum fee a prover can submit for an epoch proof quote
+
+## Oracles
+
+### `fee_asset_per_wei`
+
+This will be stored on the rollup contract. On proposing a block, proposers will be able to adjust the rate by up to `maximum_fee_asset_price_multiplier_per_block`.
+
+Has a minimum value of `minimum_fee_asset_price`.
+
+### `proving_cost_per_mana`
+
+This will be stored on the rollup contract. On proposing a block, proposers will be able to adjust the rate by up to `maximum_proving_cost_multiplier_per_block`.
+
+Has a minimum value of `minimum_proving_cost_per_mana`.
+
+### `wei_per_l1_gas` and `wei_per_l1_blob_gas`
+
+The rollup contract will maintain an `wei_per_l1_gas` oracle. 
+
+Same for `wei_per_l1_blob_gas`.
+
+They will be updated up to every `underlying_base_fee_oracle_update_interval` slots.
+
+
+## Getting the cost of a block in wei/mana
```

Comment:
I think we should be making a note around that this is the cost under the assumption that "x" mana is used. With that it is also a lot simpler to follow why some of these are constants even if not really constant.